File: archive/task06/task6.py
def get_length_of_missing_array(array_of_arrays):
    if array_of_arrays is None:
        return 0

    sorted_lens = sorted([ len(x) if x is not None else 0 for x in array_of_arrays])
    
    for i in range(len(sorted_lens) - 1):
        if sorted_lens[i] is None or sorted_lens[i] == 0:
            return 0

        if sorted_lens[i] != sorted_lens[i + 1] - 1:
            return sorted_lens[i] + 1
    
    return 0


print(get_length_of_missing_array(
    [[1, 2], [4, 5, 1, 1], [1], [5, 6, 7, 8, 9]])
    )  # 3
print(get_length_of_missing_array(
    [[5, 2, 9], [4, 5, 1, 1], [1], [5, 6, 7, 8, 9]])
    )  # 2
print(get_length_of_missing_array(
    [[None], [None, None, None]])
    )  # 2
print(get_length_of_missing_array(
    [['a', 'a', 'a'], ['a', 'a'], ['a', 'a', 'a', 'a'], ['a'], ['a', 'a', 'a', 'a','a', 'a']])
    )  # 5

File: archive/task06/task6.cpp
#include <iostream>
#include <vector>
#include <algorithm>

template<class TYPE>
int getLengthOfMissingArray(std::vector<std::vector<TYPE>> arrayOfArrays)
{
    int size = arrayOfArrays.size();
    
    std::sort(arrayOfArrays.begin(), arrayOfArrays.end(), [](auto x, auto y){ return x.size() < y.size(); });

    for (int i = 0; i < size - 1; ++i) {
        int current_size = arrayOfArrays[i].size();

        if (current_size == 0) {
            return 0;
        }

        if (current_size != arrayOfArrays[i + 1].size() - 1) {
            return current_size + 1;
        }
    }

    return 0;
}

int main() {
    std::vector<std::vector<int>> v{{1, 2}, {4, 5, 1, 1}, {1}, {5, 6, 7, 8, 9}};

    std::cout << getLengthOfMissingArray(v) << '\n';

    return 0;
}